fairscale/experimental/wgit/signal_sparsity.py

```diff
@@ -52,28 +132,81 @@ class SignalSparsity:
-        """Validating the config is valid.
+        """Validating if the config is valid.
 
-        For example, not both top_k_element and top_k_percent is set.
+        This includes asserting the following:
+        1. validating that both top_k_element and top_k_percent is not set.
+        2. Asserting that if top_k_percent is not None, it is a valid number for percentage.
 
-        this should assert fail if checking fails.
+        This asserts fail if the checking fails.
         """
-        pass
+        # assert that both top_k_elements and top_k_percent aren't set for sst and dst
+        def one_and_only(a: Optional[int], b: Optional[float]) -> bool:
+            return (a is None) ^ (b is None)
+
+        assert one_and_only(self._sst_top_k_element, self._sst_top_k_percent) and one_and_only(
+            self._dst_top_k_element, self._dst_top_k_percent
+        ), "One and only one of top_k_element and top_k_percent for each of sst and dst must be provided as argument."
+
+        # assert that, if top_k_percent is not None, it is a valid number for a percentage.
+        def none_or_in_range(a: Optional[float]) -> bool:
+            return a is None or (0.0 < a <= 100.0)
+
+        assert none_or_in_range(self._sst_top_k_percent) and none_or_in_range(
+            self._dst_top_k_percent
+        ), "top_k_percent values for sst and dst has to be in the interval (0, 100]"
 
     def dense_to_sst(self, dense: Tensor) -> Tensor:
-        """Get SST from a tensor
+        """Get Signal Sparse Tensor (SST) from a dense tensor
 
         Dense -> fft -> top-k -> results.
 
+        The input dense tensor is transformed using a transform algorithm according to the `algo`
+        initialization argument. The SST is then generated from the top_k_elements
+        (or the top_k_percentage) of values from the transformed tensor along the 'sst_top_k_dim'.
+
+        Args:
+            dense (Tensor):
+                Input dense tensor (no zeros).
+
         Returns:
-            Same shaped tensor, still in dense format but in frequency domain and has zeros.
+            Same shaped tensor as the input dense tensor, still in dense format but in frequency
+            domain (complex valued) and has zeros.
         """
-        pass
+        top_k_total_size = _top_k_total_size(dense, self._sst_top_k_dim)
+        k = _get_k_for_topk(self._sst_top_k_percent, self._sst_top_k_element, top_k_total_size)
+        dense_freq = self._transform(dense)
+
+        # NOTE: real_dense_freq can potentially be magnitude of complex frequency components
+        # or DCT transformed components when using DCT (currently not implemented).
+        # TODO: In case of the FFT, the imaginary part can perhaps be quantized or pruning can be
+        # done on the smaller phases.
+        real_dense_freq = torch.abs(dense_freq.real)
+
+        sst = _scatter_topk_to_sparse_tensor(real_dense_freq, dense_freq, k, dim=self._sst_top_k_dim)
```

Comment:
return directly?

Reply:
Was considering whether to return directly or be more explicit about what is being returned. Will return directly, thanks! :))

Reply:
The docstring should already talked about what's being returned.

A var should be introduced if it is used at least twice. If it is used only once and the original expression could be used just as well. 

In some case, the code is super unclear with the expression only. In that case, using a comment is better than trying to come up with a var name to explain the expression IMHO.

Reply:
That is very helpful, thank you :))